sqlmesh/core/snapshot/definition.py

```diff
@@ -1035,12 +1008,171 @@ def _ensure_categorized(self) -> None:
 SnapshotNameVersionLike = t.Union[SnapshotNameVersion, SnapshotTableInfo, Snapshot]
 
 
-def table_name(physical_schema: str, name: str, version: str, is_temp: bool = False) -> str:
+class DeployabilityIndex(PydanticModel, frozen=True):
```

Comment:
small nit but I'd prefer like DeployabilityMap or something else, index reminds me of array indexing

Reply:
Isn't it more like database index? We have IDs, lookup, etc.

Reply:
Sorry, I'm confused. There is neither map nor dict in there :D

Reply:
ah right